test/AggregationTest.php

```diff
@@ -192,7 +192,7 @@ function testAggregateOptimizer() {
 		$typeDay = Util\Aggregation::getAggregationLevelTypeValue('day');
 
 		// day: 2 rows of aggregation data, day first
-		$opt = $agg->getOptimalAggregationLevel(self::$uuid);
+		$opt = $agg->hasDataForAggregationLevel(self::$uuid);
```

Comment:
I'm confused that this test works. It tests for 2 result arrays but query goes only for the selected level. This should fail?!

Reply:
After doing some debugging it's obvious. There is no "selected level". If the caller does not provide a "desired" level, the generated query does not include the type column in where clause. In this case this method behaves like before.